2-repositories/iglu-server/src/main/scala/com/snowplowanalytics/iglu/server/Webhook.scala

```diff
@@ -2,29 +2,34 @@ package com.snowplowanalytics.iglu.server
 
 import cats.Applicative
 import cats.implicits._
-import com.snowplowanalytics.iglu.core.SchemaKey
+
 import io.circe.{Encoder, Json}
-import org.http4s.{Request, Uri}
+
+import org.http4s.{Request, Response, Status, Uri}
 import org.http4s.client.Client
 
+import com.snowplowanalytics.iglu.core.SchemaKey
+
 sealed trait Webhook
 
 object Webhook {
+
   case class SchemaPublished(uri: Uri, vendorPrefixes: Option[List[String]]) extends Webhook
-  case class ApiKeyPublished(uri: Uri, vendorPrefixes: Option[List[String]]) extends Webhook
 
   case class WebhookClient[F[_]](webhooks: List[Webhook], httpClient: Client[F]) {
-    def schemaPublished(schemaKey: SchemaKey, updated: Boolean)(implicit F: Applicative[F]): F[Unit] =
-      webhooks.traverse_ {
-        case SchemaPublished(uri, _) =>
+    def schemaPublished(schemaKey: SchemaKey, updated: Boolean)(implicit F: Applicative[F]): F[List[Either[String, Unit]]] =
+      webhooks.traverse {
+        case SchemaPublished(uri, prefixes) if prefixes.isEmpty || prefixes.getOrElse(List()).exists(schemaKey.vendor.startsWith(_)) =>
           val event = SchemaPublishedEvent(schemaKey, updated)
           val req = Request[F]().withUri(uri).withBodyStream(Utils.toBytes(event))
-          httpClient.fetch(req) { _ => F.pure(()) }
-
-        case ApiKeyPublished(_, _) => F.pure(())
+          httpClient.fetch(req) { res: Response[F] =>
+            res.status match {
+              case Status(code) if code != 200 => F.pure(code.toString.asLeft)
```

Comment:
I think anything withing `200-299` range can be considered as success. Also, in case server is unavailable we won't get any `Response`, we need to handle that case as well.